shell/platform/embedder/tests/embedder_test_context.h

```diff
@@ -79,8 +80,20 @@ class EmbedderTestContext {
 
   size_t GetSoftwareSurfacePresentCount() const;
 
-  // Returns the frame information for all the frames that were rendered.
-  std::vector<FlutterFrameInfo> GetGLFBOFrameInfos();
+  using GLGetFBOCallback = std::function<void(FlutterFrameInfo frame_info)>;
+
+  //----------------------------------------------------------------------------
+  /// @brief      Sets a callback that will be invoked (on the raster task
+  ///             runner) when the engine asks the embedder for a new FBO ID at
+  ///             the updated size.
+  ///
+  /// @attention  The callback will be invoked on the raster task runner. The
+  ///             callback can be set tests host thread.
+  ///
+  /// @param[in]  callback  The callback to set. The previous callback will be
+  ///                       un-registered.
+  ///
+  void SetGLGetFBOCallback(GLGetFBOCallback callback);
```

Comment:
Does this have any use outside of testing? Is there any way to make a method visible only for a test?

Reply:
`embedder/tests/embedder_test_context.h` is a test only class.